pkg/fileio/file_io.go

```diff
@@ -0,0 +1,61 @@
+package fileio
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"text/template"
+)
+
+func WriteFile(filename string, contents string, templateData any) error {
+	if templateData == nil {
+		if err := os.WriteFile(filename, []byte(contents), os.ModePerm); err != nil {
+			return fmt.Errorf("writing file: %w", err)
```

Comment:
I'm not familiar with what the output will be if this error happens, but I like to include the name of the file so there's some added context in the log message:

```
return fmt.Errorf("writing file %s: %w", filename, err)
```

Reply:
Leaving for completeness, but Atanas and I discussed this in chat and his approach is to leave it to the caller to mention the filename.